src/pages/EntryPage.vue

```diff
@@ -39,32 +39,30 @@ const shareStore = useShareStore()
 
 const tab = ref(entryStore.tab)
 
+entryStore.fetchEntries().catch((error) => errorStore.throwError(error))
+
 const entry = computed(() => {
   return entryStore.getEntries?.find((entry) => entry.slug === router.currentRoute.value.href)
 })
+const myTimeout = ref()
 
-onMounted(async () => {
-  await entryStore.fetchEntries().catch((error) => errorStore.throwError(error))
-
-  if (!entry.value?.id) {
-    await new Promise((resolve) => setTimeout(resolve, 2000)) // wait 2 seconds before continue
-  }
-
+//Handle 404
+myTimeout.value = setTimeout(() => {
   if (!entry.value?.id) {
     router.push('/404')
-    return
   }
+}, 30000)
```

Comment:
This is interesting code... why would there be up to a 3 sec delay before the  entry.value.id was populated?

Reply:
This is 30 seconds.